### PR TITLE
Gh7075

### PR DESCRIPTION
**Pull request by pvpurcarcosmin**

### CODE REVIEW

**Comment by pvpurcarcosmin** [comment]
🔴 🔴 🔴  PLS don't merge! I'll do it as I need to fix commit messages!

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/candidate/commands/ProcessHUsAndPickingCandidateCommand.java [hunk 1]
```diff
@@ -138,23 +138,39 @@ private void allocateHUsToShipmentSchedule()
...
+
+		getPickingCandidatesForHUId(huId)
 				.stream()
 				.map(this::createPackingItemPart)
-				.collect(PackingItemParts.collect());
-
-		return PackingItems.newPackingItem(parts);
+				.map(PackingItems::newPackingItem)
+				.forEach(item -> {
+					final IPackingItem existingItem = packingItems.get(item.getGroupingKey());
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/candidate/commands/ProcessHUsAndPickingCandidateCommand.java:161]: u can use Map.merge

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_M_Picking_Candidate_Process.java [hunk 2]
```diff
@@ -148,4 +154,18 @@ protected void postProcess(final boolean success)
...
+		Check.assume(pickingSlotRowOrHU.isPickedHURow(), "Was expecting an HuId but found none!");
+
+		if (pickingSlotRowOrHU.getHuQtyCU() != null && pickingSlotRowOrHU.getHuQtyCU().signum() > 0)
+		{
+			return false;
+		}
+
+		final I_M_HU hu = handlingUnitsBL.getById(pickingSlotRowOrHU.getHuId());
+
+		return handlingUnitsBL.getStorageFactory().streamHUProductStorages(hu).allMatch(IProductStorage::isEmpty);
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_M_Picking_Candidate_Process.java:169]: refactor this and introduce IHandlingUnits.isEmptyStorage(I_M_HU)

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java [hunk 3]
```diff
@@ -135,4 +129,76 @@ public Object getParameterDefaultValue(@NonNull final IProcessDefaultParameter p
...
+		}
+
+		final Set<ShipmentScheduleId> shipmentScheduleIds = getPickingCandidateService().getScheduleIdsByHuId(pickingSlotRow.getHuId());
+
+		if (shipmentScheduleIds.isEmpty())
+		{
+			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_NO_PICKED_HU_FOUND));
+		}
+
+		final ImmutableSet<OrderId> orderIds = getShipmentSchedulePA().getByIds(shipmentScheduleIds)
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java:151]: refactor this in a new method, i.e. extractOrderIds(Set<ShipmentScheduleId>)

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java [hunk 4]
```diff
@@ -135,4 +129,76 @@ public Object getParameterDefaultValue(@NonNull final IProcessDefaultParameter p
...
+		{
+			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_NO_PICKED_HU_FOUND));
+		}
+
+		if (orderIds.size() > 1)
+		{
+			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_TO_THE_SAME_HU_FOR_MULTIPLE_ORDERS_NOT_ALLOWED));
+		}
+
+		final OrderId orderIdOfThePickedHU = orderIds.asList().asList().get(0);
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java:169]: only one asList() is enough :)

**Comment by pvpurcarcosmin** [comment]
🔴 🔴 🔴 PLS don't merge! I'll do it as I need to fix commit messages!

**Pull request merged**